org.idempiere.test/src/org/idempiere/test/base/DocumentStatusTest.java

```diff
@@ -0,0 +1,144 @@
+				USER_GARDENADMIN, ROLE_GARDENWORLDADMIN, getTrxName());
+		assertTrue(Arrays.asList(documentStatusIndicators).contains(invoiceDS), "Either User or Role Assignment matching");
+		assertTrue(Arrays.asList(documentStatusIndicators).contains(noticeDS), "No permission assigned, should visible to every one");
+		assertFalse(Arrays.asList(documentStatusIndicators).contains(requestDS),
+				"When Role not matching, Should not Visible");
+		assertTrue(Arrays.asList(documentStatusIndicators).contains(WorkflowActivitiesDS),
+				"User is assigned, Should Visible");
+
+		// check document status accessibility with user- GardenAdmin and role- GardenWorld User
+		documentStatusIndicators = MDocumentStatus.getDocumentStatusIndicators(Env.getCtx(), USER_GARDENADMIN,
+				ROLE_GARDENWORLDUSER, getTrxName());
+		assertFalse(Arrays.asList(documentStatusIndicators).contains(invoiceDS),
+				"User Assignment Match but role do not");
+		assertTrue(Arrays.asList(documentStatusIndicators).contains(noticeDS), "No permission assigned, should visible to every one");
+		assertTrue(Arrays.asList(documentStatusIndicators).contains(requestDS), "When Role matching, Should  Visible");
+		assertTrue(Arrays.asList(documentStatusIndicators).contains(WorkflowActivitiesDS),
+				"User is assigned, Should Visible");
+
+		// check document status accessibility with user- GardenUser and role- GardenWorld User
+		documentStatusIndicators = MDocumentStatus.getDocumentStatusIndicators(Env.getCtx(), USER_GARDENUSER,
+				ROLE_GARDENWORLDUSER, getTrxName());
+		assertFalse(Arrays.asList(documentStatusIndicators).contains(invoiceDS),
+				"Invoice Document Status Not Visible For User Garden Admin and Role GardenWorld User");
+		assertTrue(Arrays.asList(documentStatusIndicators).contains(noticeDS), "No permission assigned, should visible to every one");
+		assertTrue(Arrays.asList(documentStatusIndicators).contains(requestDS), "When Role not matching, Should not Visible");
+		assertFalse(Arrays.asList(documentStatusIndicators).contains(WorkflowActivitiesDS),
+				"User is assigned, Should not Visible");
+
+		// check document status accessibility with user- SuperUser and role- GardenWorld Admin Not Advanced
+		documentStatusIndicators = MDocumentStatus.getDocumentStatusIndicators(Env.getCtx(), USER_SUPERUSER,
+				ROLE_GARDENWORLD_ADMIN_NOT_ADVANCED, getTrxName());
+		assertFalse(Arrays.asList(documentStatusIndicators).contains(invoiceDS),
+				"Niether User or Role Assignment match, Should not visible");
+		assertTrue(Arrays.asList(documentStatusIndicators).contains(noticeDS), "No permission assigned, should visible to every one");
+		assertFalse(Arrays.asList(documentStatusIndicators).contains(requestDS),
+				"Role assignment not matching, Should not visible");
+		assertFalse(Arrays.asList(documentStatusIndicators).contains(WorkflowActivitiesDS),
+				"User assignment not visible, Should not Visible");
+
+		rollback();
```

Comment:
you can have it here but it is not really needed, the default for each unit test method is always rollback.